pkg/odo/util/completion/completion.go

```diff
@@ -48,7 +48,24 @@ func NewParsedArgs(args complete.Args, cmd *cobra.Command) parsedArgs {
 	typed := getUserTypedCommands(args, cmd)
 	commands, flagValues := getCommandsAndFlags(typed, cmd)
 
-	complete.Log("Parsed flag values: %v", flagValues)
+	complete.Log("Parsed commands values for full input: %v", commands)
+	complete.Log("Parsed flag values for full input: %v", flagValues)
+
+	// this part is meant to handle incomplete flags when other input already exists,
+	// so 2 tokens is the minimum that makes sense
+	if len(typed) > 2 {
+		typedWithoutLast := make([]string, len(typed))
```

Comment:
Could you comment the logic here some more, please?

Reply:
Sure thing

Reply:
Done